edbdeploy/projects/virtualbox.py

```diff
@@ -0,0 +1,505 @@
+        with AM("Checking instances availability"):
+            vagrant.up()
+        # Build ip address list for virtualbox deployment
+        with AM("Build VirutalBox Ansible IP addresses"):
+            # Assigning Reference Architecture
+            self.env.reference_architecture = \
+                self.ansible_vars['reference_architecture']
+
+            # Load specifications
+            self.env.cloud_spec = self._load_cloud_specs(self.env)
+
+            # Build virtualbox Ansible IP addresses
+            self._build_virtualbox_ips(self.env)
+
+    def create(self):
+        # Overload Project.create() by creating project directory only
+        with AM("Creating project directory %s" % self.project_path):
+            os.makedirs(self.project_path)
+
+    def check_versions(self):
+        # Overload Project.check_versions()
+        # Check Ansible version
+        ansible = AnsibleCli('dummy', bin_path=self.cloud_tools_bin_path)
+        ansible.check_version()
+
+        # Update before committing with
+        # projects_root_path
+        self.vagrant_project_path = os.path.join(
+            self.projects_root_path,
+            'virtualbox',
+            self.name
+        )
+        # Check only Python3 version when working with virtualbox deployment
+        vm = VirtualBoxCli(
+            'dummy', self.name, self.cloud, 0, 0, self.vagrant_project_path,
+            bin_path=self.cloud_tools_bin_path
+        )
+        vm.python_check_version()
+        vm.vagrant_check_version()
+
+    def overwrite_vagrant_params(self, mem_size, cpu_count):
+        vagrantfile = os.path.join(
+            self.vagrant_project_path,
+            'Vagrantfile'
+            )
+        with AM("Overwriting default variables in Vagrantfile %s" % vagrantfile):
+            with open(vagrantfile, 'r') as file :
+                vagrantfiledata = file.read()
+
+            # Overwrites default values memory size and cpu count in Vagrantfile
+            vagrantfiledata = vagrantfiledata.replace('v.memory = 3072', 'v.memory = %s' % mem_size)
+            vagrantfiledata = vagrantfiledata.replace('v.cpus = 2', 'v.cpus = %s' % cpu_count)
+
+            # Write the Vagrantfile out again
+            with open(vagrantfile, 'w') as file:
+                file.write(vagrantfiledata)
+
+    def provision(self, env):
+        # Overload Project.provision()
+        self._load_ansible_vars()
+
+        # Update before committing with
+        # projects_root_path
+        self.vagrant_project_path = os.path.join(
+            self.projects_root_path,
+            'virtualbox',
+            self.name,
+        )
+        mem_size = self.ansible_vars['mem_size']
+        cpu_count = self.ansible_vars['cpu_count']
+        vagrant = VirtualBoxCli(
+            self.cloud, self.name, self.cloud, mem_size, cpu_count,
+            self.vagrant_project_path, bin_path=self.cloud_tools_bin_path
+        )
+        with AM("Checking instances availability"):
```

Comment:
Maybe we could update action's name here?

Reply:
Changed it to `Provisioning Virtual Machines`